client/src/components/AuthContext.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useMemo, useState } from 'react';
+import { accessTokenService } from '../services/accessTokenService.js';
+import { authService } from '../services/authService.js';
+
+export const AuthContext = React.createContext({});
+
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [isChecked, setChecked] = useState(true);
+
+  async function activate(activationToken) {
+    const { accessToken, user } = await authService.activate(activationToken);
+
+    accessTokenService.save(accessToken);
+    setUser(user);
+  }
+
+  async function checkAuth() {
+    try {
+      const { accessToken, user } = await authService.refresh();
+
+      accessTokenService.save(accessToken);
+      setUser(user);
+    } catch (error) {
+      console.log('User is not authentincated');
```

Comment:
There is a typo in the log message. It should be 'authenticated' instead of 'authentincated'.